tests/memberships/test_membership_services.py

```diff
@@ -0,0 +1,176 @@
+
+
+@pytest.fixture(scope="module")
+def another_user(app):
+    """Community owner user."""
+    return create_test_user('another_user@example.com')
+
+
+@pytest.fixture(scope="module")
+def community_owner_identity(community_owner):
+    """Simple identity fixture."""
+    owner_id = community_owner.id
+    i = Identity(owner_id)
+    i.provides.add(UserNeed(owner_id))
+    i.provides.add(Need(method="system_role", value="any_user"))
+    i.provides.add(Need(method="system_role", value="authenticated_user"))
+    return i
+
+
+@pytest.fixture(scope="module")
+def another_identity(another_user):
+    """Simple identity fixture."""
+    user_id = another_user.id
+    i = Identity(user_id)
+    i.provides.add(UserNeed(user_id))
+    i.provides.add(Need(method="system_role", value="any_user"))
+    i.provides.add(Need(method="system_role", value="authenticated_user"))
+    return i
+
+
+@pytest.fixture(scope="module")
+def community_creation_input_data():
+    """Full community data used as input to community service."""
+    return  {
+        "access": {
+            "visibility": "public",
+            "member_policy": "open",
+            "record_policy": "open",
+        },
+        "id": "my_community_id",
+        "metadata": {
+            "title": "My Community",
+            "description": "This is an example Community.",
+            "type": "event",
+            "curation_policy": "This is the kind of records we accept.",
+            "website": "https://inveniosoftware.org/",
+            "organizations": [{
+                    "name": "CERN",
+            }]
+        }
+    }
+
+
+@pytest.fixture(scope="function")
+def community_service(app, location):
+    """Community service.
+
+    Snuck in the location fixture, because needed on community creation
+    i.e. almost every time this service is used.
+    """
+    return current_communities.service
+
+
+@pytest.fixture(scope="module")
+def requests_service(app):
+    """Requests service."""
+    return current_requests_service
+
+
+@pytest.fixture(scope="function")
+def community(community_service, community_owner_identity,
+              community_creation_input_data):
+    """Community fixture."""
+    return community_service.create(
+        community_owner_identity, community_creation_input_data
+    )
+
+
+@pytest.fixture()
+def invitation_creation_input_data(another_user, community):
+    """Full invitation data used as input to invitation service."""
+    return {
+        "type": "community-member-invitation",
+        "receiver": {"user": another_user.id},
+        "topic": {'community': community.data['uuid']}
+    }
+
+
+# Tests
+
+def test_invite_user_flow(
+        another_identity, community, community_owner_identity,
+        community_service, invitation_creation_input_data,
+        requests_service):
+    community_id = community.data['uuid']
+
+    # Invite
+    invitation = community_service.invitations.create(
```

Comment:
Something to note (maybe a smell), we are using 
- the invitations service to create an invitation
- the requests service to accept it
- the members service to check the final membership

It makes sense... but maybe we are just leaking the underlying abstractions...